### PR TITLE
implement auto login

### PR DESCRIPTION
**Pull request by vedi**
@refaelos, waiting for your code review

### CODE REVIEW

File: SoomlaiOSProfile/AuthController.m [hunk 1]
```diff
@@ -132,5 +131,53 @@ - (BOOL)tryHandleOpenURL:(NSURL *)url sourceApplication:(NSString *)sourceApplic
...
+
+- (void)performAutoLogin {
+    for (id key in self.providers) {
+        id<IAuthProvider> authProvider = self.providers[key];
+        Provider provider = [authProvider getProvider];
+        if ([self wasLoggedInWithProvider:provider]) {
+            NSString *payload = @"";
+            Reward *reward = nil;
+            if ([authProvider isLoggedIn]) {
+                [self setLoggedInForProvider:provider toValue:NO];
```
Comment (borisspektor) [inline SoomlaiOSProfile/AuthController.m]: Why do we need this line?

File: SoomlaiOSProfile/AuthController.m [hunk 2]
```diff
@@ -132,5 +131,53 @@ - (BOOL)tryHandleOpenURL:(NSURL *)url sourceApplication:(NSString *)sourceApplic
...
+
+- (void)afterLoginWithAuthProvider:(id <IAuthProvider>)authProvider withReward:(Reward *)reward withPayload:(NSString *)payload {
+    [authProvider getUserProfile:^(UserProfile *userProfile) {
+        [UserProfileStorage setUserProfile:userProfile];
+        if (reward) {
+            [reward give];
+        }
+
+        [self setLoggedInForProvider:[authProvider getProvider] toValue:YES];
+        [ProfileEventHandling postLoginFinished:userProfile withPayload:payload];
```
Comment (borisspektor) [inline SoomlaiOSProfile/AuthController.m]: Why not change the order here to Event->Reward as we discussed?

File: SoomlaiOSProfile/AuthController.m [hunk 3]
```diff
@@ -22,8 +22,12 @@
 #import "IAuthProvider.h"
 #import "Reward.h"
 #import "SoomlaUtils.h"
+#import "KeyValueStorage.h"
 
-@implementation AuthController
+@implementation AuthController {
+}
+
+static NSString* DB_KEY_PREFIX  = @"soomla.profile.common";
```
Comment (borisspektor) [inline SoomlaiOSProfile/AuthController.m]: maybe have something in the key to indicate it's being used for login?

**Comment by vedi** [comment]
@refaelos, @borisspektor, applied code review comments, waiting for your feedback

**Pull request merged**